src/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,177 @@
+import React, {
+  PropsWithChildren,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react'
+import styled from 'styled-components'
+
+import Grid from '../Grid'
+
+export interface DrawerProps {
+  defaultIsOpen?: boolean
+  title?: string
+  width?: string
+  color?: string
+  background?: string
+}
+
+const Drawer: React.FC<PropsWithChildren<DrawerProps>> = (props) => {
+  const [isOpen, setIsOpen] = useState(props.defaultIsOpen || false)
+  const contentRef: React.RefObject<HTMLDivElement> | null = useRef(null)
+
+  const clickHandler = useCallback(() => {
+    setIsOpen(!isOpen)
+  }, [isOpen, setIsOpen])
+
+  const title = props.title || ''
+  const width = props.width || '100%'
+  const color = props.color || 'black'
+  const background = props.background || 'transparent'
+  const contentHeight = contentRef.current?.clientHeight || 1000
+
+  return (
+    <DrawerWrapper width={width}>
+      <DrawerHeader
+        title={title}
```

Comment:
i think getting picky but I follow different e.g. Drawer header could take ..props
and component handle default, i found it easier to read
```jsx

const DrawerHeader: React.FC<DrawerHeaderProps> = ({ title = '', color = 'black' }) => {
  return (
    <DrawHeade
```